k8s/distributed-chroma/templates/compaction-service.yaml

```diff
@@ -29,28 +29,42 @@ spec:
         member-type: compaction-service
     spec:
       serviceAccountName: compaction-service-serviceaccount
-      {{if .Values.compactionService.configuration}}
       volumes:
+        {{if .Values.compactionService.configuration}}
         - name: compaction-service-config
           configMap:
             name: compaction-service-config
-      {{ end }}
+        {{ end }}
+        {{if .Values.compactionService.cache}}
+        - name: compaction-service-cache
+          hostPath:
+            path: {{ .Values.queryService.cache.hostPath }}
+            type: DirectoryOrCreate
+        {{ end }}
       containers:
         - name: compaction-service
           image: "{{ .Values.compactionService.image.repository }}:{{ .Values.compactionService.image.tag }}"
           imagePullPolicy: IfNotPresent
-          {{if .Values.compactionService.configuration}}
           volumeMounts:
+            {{if .Values.compactionService.configuration}}
             - name: compaction-service-config
               mountPath: /config/
-          {{ end }}
+            {{ end }}
+            {{if .Values.compactionService.cache}}
+            - name: compaction-service-cache
+              mountPath: {{ .Values.compactionService.cache.mountPath }}
+            {{ end }}
           ports:
             - containerPort: 50051
           env:
             {{if .Values.compactionService.configuration}}
             - name: CONFIG_PATH
               value: /config/config.yaml
             {{ end }}
+            {{if .Values.compactionService.cache}}
+            - name: CACHE_PATH
```

Comment:
Should we make the CACHE_PATH an env var override so that the config is the single source of truth ?

Reply:
Can we do away with the env var and rely upon our configs?  The only thing it can do is drift from the config.

Reply:
I'm not sure how the config could be the single source of truth. We'd have to read the value out of the config or its associated configMap to do the mount unless I'm misunderstanding something.

Reply:
> Can we do away with the env var and rely upon our configs? The only thing it can do is drift from the config.

Just to be clear, you're saying that whatever we set the cache path to in `config.yaml` should therefore be configured as a volume mount, correct? If so, the helm chart template here is going to need to know about the contents of the `config.yaml` in order to specify the correct `volumes` and `volumeMounts` configuration, unless I'm missing something? I'm not sure how that would work with helm.

Reply:
Until I have more clarity, I suppose the last thing I'll mention is that this `CACHE_PATH` env setting should probably just go away since it's confusing as to who should be using it and whatnot. 

IMHO, the system (aka chart) should be making a decision as to where a host-mounted volume is provided to the container and then the code, should then be configured accordingly. If those two aren't in agreement, then the application won't be configured correctly. 

Does this make sense to everyone?

Reply:
It does.  I was simply suggesting that having the cache_path variable would be confusing since the code always pulls the path from the config.  Would you find it more intuitive to have CACHE_PATH always be set and used?